code/modules/mob/living/basic/bots/_bots.dm

```diff
@@ -0,0 +1,781 @@
+
+/mob/living/basic/bot/proc/check_access(mob/living/user, obj/item/card/id)
+	if(!istype(user)) // Non-living mobs shouldn't be manipulating bots (like observes using the botkeeper UI).
+		return FALSE
+	if(user.has_unlimited_silicon_privilege || isAdminGhostAI(user)) // Silicon and Admins always have access.
+		return TRUE
+	if(!length(maints_access_required)) // No requirements to access it.
+		return TRUE
+	if(bot_access_flags & BOT_CONTROL_PANEL_OPEN) // Unlocked.
+		return TRUE
+
+	var/obj/item/card/id/used_id = id || user.get_idcard(TRUE)
+
+	if(!used_id || !used_id.access)
+		return FALSE
+
+	for(var/requested_access in maints_access_required)
+		if(requested_access in used_id.access)
+			return TRUE
+
+	return FALSE
+
+/mob/living/basic/bot/bee_friendly()
+	return TRUE
+
+/mob/living/basic/bot/death(gibbed)
+	if(paicard)
+		ejectpai()
+	explode()
+	return ..()
+
+/mob/living/basic/bot/proc/explode()
+	visible_message(span_boldnotice("[src] blows apart!"))
+	do_sparks(3, TRUE, src)
+	var/atom/location_destroyed = drop_location()
+	if(prob(50))
+		drop_part(robot_arm, location_destroyed)
+
+/mob/living/basic/bot/emag_act(mob/user, obj/item/card/emag/emag_card)
+	. = ..()
+	if(!(bot_access_flags & BOT_CONTROL_PANEL_OPEN)) //First emag application unlocks the bot's interface. Apply a screwdriver to use the emag again.
+		bot_access_flags |= BOT_CONTROL_PANEL_OPEN
+		balloon_alert(user, "cover unlocked")
+		return TRUE
+	if(!(bot_access_flags & BOT_CONTROL_PANEL_OPEN) || !(bot_access_flags & BOT_MAINTS_PANEL_OPEN)) //Bot panel is unlocked by ID or emag, and the panel is screwed open. Ready for emagging.
+		balloon_alert(user, "open maintenance panel first!")
+		return FALSE
+	bot_access_flags |= BOT_COVER_EMAGGED
+	bot_access_flags &= ~BOT_CONTROL_PANEL_OPEN
+	bot_mode_flags &= ~BOT_MODE_REMOTE_ENABLED //Manually emagging the bot also locks the AI from controlling it.
+	bot_reset()
+	turn_on() //The bot automatically turns on when emagged, unless recently hit with EMP.
+	to_chat(src, span_userdanger("(#$*#$^^( OVERRIDE DETECTED"))
+	to_chat(src, span_boldnotice(get_emagged_message()))
+	if(user)
+		log_combat(user, src, "emagged")
+	return TRUE
+
+/mob/living/basic/bot/examine(mob/user)
+	. = ..()
+	if(health < maxHealth)
+		if(health > maxHealth/3)
+			. += "[src]'s parts look loose."
+		else
+			. += "[src]'s parts look very loose!"
+	else
+		. += "[src] is in pristine condition."
+
+	. += span_notice("Its maintenance panel is [bot_access_flags & BOT_MAINTS_PANEL_OPEN ? "open" : "closed"].")
+	. += span_info("You can use a <b>screwdriver</b> to [bot_access_flags & BOT_MAINTS_PANEL_OPEN ? "close" : "open"] it.")
+
+	if(bot_access_flags & BOT_MAINTS_PANEL_OPEN)
+		. += span_notice("Its control panel is [bot_access_flags & BOT_CONTROL_PANEL_OPEN ? "unlocked" : "locked"].")
+		var/is_sillycone = issilicon(user)
+		if(!(bot_access_flags & BOT_COVER_EMAGGED) && (is_sillycone || user.Adjacent(src)))
```

Comment:
```suggestion
		if(!(bot_access_flags & BOT_COVER_EMAGGED) && (issilicon(user) || user.Adjacent(src)))
```
No point in it

Reply:
it was caching it because it uses this check again in the span info!